presto-main/src/test/java/io/prestosql/sql/planner/TestLogicalPlanner.java

```diff
@@ -1105,4 +1108,37 @@ public void testRedundantDistinctLimitNodeRemoval()
                                         node(ProjectNode.class,
                                                 values(ImmutableList.of("x")))))));
     }
+
+    @Test
+    public void testHashGenerationOptimizer()
```

Comment:
- the name is too generic. Maybe `testHashComputationForAggregationOnTopOfUnion`?
- what about test for mark distinct?